### PR TITLE
[CodingStyle] Use json_encode for json encode if the class 'Nette\Utils\Json' not found

### PR DESCRIPTION
**Pull request by zingimmick**
Fixes rectorphp/rector#6637

### CODE REVIEW

File: rules/CodingStyle/NodeFactory/JsonEncodeStaticCallFactory.php [hunk 1]
```diff
@@ -34,7 +36,11 @@ public function createFromArray(Expr $assignExpr, Array_ $jsonArray): Assign
         $jsonDataAssign = new Assign($assignExpr, $jsonArray);
 
         $jsonDataVariable = new Variable('jsonData');
-        $jsonDataAssign->expr = $this->nodeFactory->createStaticCall('Nette\Utils\Json', 'encode', [$jsonDataVariable]);
+        if ($this->reflectionProvider->hasClass('Nette\Utils\Json')) {
+            $jsonDataAssign->expr = $this->nodeFactory->createStaticCall('Nette\Utils\Json', 'encode', [$jsonDataVariable]);
+        } else {
+            $jsonDataAssign->expr = $this->nodeFactory->createFuncCall('json_encode', [$jsonDataVariable]);
+        }
```
Comment (TomasVotruba) [inline rules/CodingStyle/NodeFactory/JsonEncodeStaticCallFactory.php]: Thank you for the PR :+1: the solution looks very well.

The name of class `JsonEncodeStaticCallFactory`  suggests this must create a `StaticCall`. Anyting else is unexpected.

This should be resolved in the rule level:

https://github.com/rectorphp/rector/blob/d3c644f12fa572c6a4d047c18113f0a9b516a6f2/rules/CodingStyle/Rector/Assign/ManualJsonStringToJsonEncodeArrayRector.php#L111

Comment (zingimmick) [reply rules/CodingStyle/NodeFactory/JsonEncodeStaticCallFactory.php]: Moved to the `ManualJsonStringToJsonEncodeArrayRector`

**Comment by TomasVotruba** [comment]
Thank you :+1:

File: rules/CodingStyle/Rector/Assign/ManualJsonStringToJsonEncodeArrayRector.php [hunk 2]
```diff
@@ -294,4 +297,17 @@ private function getNextExpression(Node $node): ?Node
...
+
+    private function createJsonEncodeAssign(Expr $assignExpr, Array_ $jsonArray): Assign
+    {
+        if ($this->reflectionProvider->hasClass('Nette\Utils\Json')) {
+            return $this->jsonEncodeStaticCallFactory->createFromArray($assignExpr, $jsonArray);
+        }
+
+        $jsonDataAssign = new Assign($assignExpr, $jsonArray);
+        $jsonDataVariable = new Variable('jsonData');
+        $jsonDataAssign->expr = $this->nodeFactory->createFuncCall('json_encode', [$jsonDataVariable]);
```
Comment (staabm) [inline rules/CodingStyle/Rector/Assign/ManualJsonStringToJsonEncodeArrayRector.php:309]: To mimic nette-json more closely, should this use e.g. `JSON_THROW_ON_ERROR` etc ?

see https://github.com/nette/utils/blob/773238807e7aaa9fe354908a32955076972af735/src/Utils/Json.php#L36 which flags nette uses by default

Comment (TomasVotruba) [reply rules/CodingStyle/Rector/Assign/ManualJsonStringToJsonEncodeArrayRector.php:309]: I think this is not to mimic nette-json, but rather original behavior.
The added value in `Nette\Utils\Strings` (including all the error handling) is just a bonus.

**Pull request merged**